sim/simulator.py

```diff
@@ -107,6 +180,7 @@ def restart(self, alg: Algorithm):
             raise RuntimeError(f"robot malfunction - yielded {state} instead of {Action.READY}")
         self._status = SimulationStatus.READY
         print("sim: robot is ready")
+        self._counter = Counter()
```

Comment:
If you're not creating the `self._counter` in the `__init__`, do the same as the other fields and put an annotation for it.